vignettes/data-and-qc.Rmd

```diff
@@ -0,0 +1,125 @@
+  getSampleTable() %>% 
+  dplyr::select(sample_name, input_relH)
+```
+
+We typically see values of the `input_relH` between 1.3-1.6, suggesting that there are more CGs seen in the sequenced fragments than would be expected from the genomic sequence. This is presumably due to the evolutionary suppression of `CG` dinucleotides outside of specific contexts, meaning they are less common in those areas of the genome that are repetitive and hence less easy to map. This is likely to be dependent on the particular sequencing methods and the sample type under consideration. We therefore need to see an increase between the enriched and non-enriched data:
+
+```{r}
+exampleTumourNormal %>% 
+  addLibraryInformation() %>% 
+  getSampleTable() %>% 
+  dplyr::select(sample_name, relH_MeCap = relH, relH_Input = input_relH) %>% 
+  pivot_longer(matches("relH"), 
+               names_prefix = "relH", 
+               names_to = "enrichment", 
+               values_to = "relH") %>%
+  ggplot(aes(x = sample_name, y = relH, col = enrichment)) +
+  geom_point() +
+  theme_bw()
+```
+
+There is a second relative enrichment metric defined by the `MEDIPs` package, known as `GoGe`, that is also calculated globally in a similar way but considers the ratio between the number of `CG`s present in the fragments compared to the number of `C`s and `G`s independently. This is calculated and stored in the `libraries` information alongside the `relH` metric.
+
+## Fragments without pattern
+The `libraries` tab also calculates how many fragments do not contain a CG at all. This can be measure of background noise.
+
+## Hyper-stably methylated regions
+The other metric that we typically use for quality control is to consider windows that are known to be commonly methylated across a range of tissues and diseases. For instance, for the human genome, [Edgar et al (2014)]((https://pubmed.ncbi.nlm.nih.gov/25493099/) identified probes from the Illumina 450k methylation array that were consistently methylated in over 1000 samples. We therefore take the windows covered by these probes, and calculate beta values for them, followed by determining how many of these windows have a calculated beta value above a threshold (e.g. 0.8). As noted above, beta values will not be calculated if a sample does not have sufficient coverage and/or enrichment, meaning this metric incorporates both the coverage depth as well as how well the enrichment has worked. For our samples within the National Biomarker Centre, we currently require at least 40% of the hyperstable windows with a `CpG_density` above 5 to have a beta value above 0.8. Samples with lower enrichment scores (as described by the `relH` score) require a greater coverage depth to reach this threshold.
+
+For hg38, these hyper-stable methylated probes are incorporated in a data object `hg38UltraStableProbes`, and a function `addHyperStableFraction` is provided to calculate this and add as a column to the sampleTable for filtering.
```

Comment:
It might be helpful mentioning that this will not work with the example data as there aren't enough regions, in case someone wants to try it.